src/engine/SCons/Platform/__init__.py

```diff
@@ -205,12 +205,11 @@ def __call__(self, target, source, env, for_signature):
         # We use the .lnk suffix for the benefit of the Phar Lap
         # linkloc linker, which likes to append an .lnk suffix if
         # none is given.
-        if env.has_key('TEMPFILESUFFIX'):
-            suffix = env.subst('$TEMPFILESUFFIX')
-        else:
-            suffix = '.lnk'  # TODO: better way to pick default?
-        fd, tmp = tempfile.mkstemp(suffix, text=True)
+        suffix = env.subst('$TEMPFILESUFFIX')
+        if not suffix:
```

Comment:
I'd revert this bit. If the user sets TEMPFILESUFFIX='' on purpose, it get's overwritten.